### PR TITLE
Add mig-controller e2e tests

### PR DESCRIPTION
**Pull request by fbladilo**
- Create ansible assets to cover all current mig-controller sample scenarios

### CODE REVIEW

**Comment by fbladilo** [comment]
This covers all the mig-controller current test scenarios : 

1. NFS-PV
2. Stateless (nginx)
3. Local-images (s2i images transfer from cluster to cluster)

Instructions :

File: roles/nfs-pv/tasks/prepare.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  stat:
+    path: "{{ playbook_dir }}/mig-controller"
+  register: repo
+
+- name: Clone the mig-controller repo
+  git:
+    repo: 'https://github.com/fusor/mig-controller.git'
+    dest: "{{ playbook_dir }}/mig-controller"
+    force: true
+  when: not repo.stat.exists
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply roles/nfs-pv/tasks/prepare.yml]: Already fixed in latest updates.

File: roles/nfs-pv/tasks/prepare.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+- name: Check that mig-controller repo exists
+  stat:
+    path: "{{ playbook_dir }}/mig-controller"
+  register: repo
+
+- name: Clone the mig-controller repo
+  git:
+    repo: 'https://github.com/fusor/mig-controller.git'
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply roles/nfs-pv/tasks/prepare.yml]: Was also fixed.

File: roles/local-images/tasks/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+- name: Create namespace for local-images
+  k8s:
+    name: "{{ namespace }}"
+    api_version: v1
+    kind: Namespace
+    state: present
+
+- name: Build sample new-app using s2i
+  shell: "oc -n {{ namespace }} new-app {{ s2i_app_location }}"
```
Comment (Danil-Grigorev) [inline roles/local-images/tasks/deploy.yml]: Just noticed, may you please change the `oc` to `{{ oc_binary | default('oc') }}`? We need it badly on CI side.

File: roles/local-images/tasks/check.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  until: "true in (pod | json_query('resources[].status.containerStatuses[].ready'))"
+  retries: 10
+  delay: 15
+  with_items:
+   - "test-dc"
+   - "test-job"
+   - "test-standalone"
+   - "test-deployment"
+   - "test-statefulset"
+   - "test-daemonset"
```
[the remark this answers is not in the document]
Comment (fbladilo) [reply roles/local-images/tasks/check.yml]: :+1:

File: roles/local-images/tasks/deploy.yml (hunk 3 above)
Comment (fbladilo) [reply roles/local-images/tasks/deploy.yml]: I will address this one on a subsequent PR, I want to sync with mig-ci in how we address oc_binary location...

**Pull request merged**